core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4179,6 +4179,55 @@ public static Map strToMap(String string, String stringDelimiter, String keyValu
     return map;
   }
 
+  /** Support the SUBSTRING_INDEX function. */
+  public static String substringIndex(String string, String delimiter, int count) {
```

Comment:
The implementation reference： https://github.com/apache/spark/blob/f7e8da3fb6dba44ab339a095c46cc8872d83b741/common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java#L819